### PR TITLE
Optionally Name all Elements, Add Marker

### PR DESCRIPTION
**Pull request by ax3l**
- [x] New Element: `Marker(name="user-specified")`
- [x] Ensure overhead on GPU start is always only 8 byte (1 pointer copy)
- [x] Unify `Marker` and `Empty` element? No.
  - in our variant, this only works if we make `name` an optional argument (so we can default-construct the first element in `KnownElementsList`), which is a bit counter-intuitive for a marker.
  - discussed: we want to have a mandatory label on each marker.
- [x] Add a user-defined name to each element (last argument, optional)

## Inputs file Usage

No changes. We remember the left-hand-side, user-defined "element name" now internally, too.

## Python Usage

No breaking change. New optional argument `name` for each element (besides the `Emtpy` one).

```py
Drift(ds=1.0)
```
The last argument to all elements is now the argument `name`.
```py
Drift(ds=1.0, name="drift1")
```

It can also be pulled forward with named arguments, e.g.,
```py
Drift(name="drift1", ds=1.0)
```

The default is `None` (no name).

Names do _not_ need to be unique at the moment, they are just labels that we will use for visualization and more descriptive book keeping.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -28,7 +28,7 @@
     // it here so users can pass command line arguments
     impactx::initialization::default_init_AMReX(argc, argv);
 
-    BL_PROFILE_VAR("main()", pmain);
+    //BL_PROFILE_VAR("main()", pmain);
```
[the remark this answers is not in the document]
Comment (ax3l) [reply src/main.cpp]: unrelated, removed.

File: src/main.cpp [hunk 2]
```diff
@@ -37,7 +37,7 @@
         impactX.evolve();
         impactX.finalize();
     }
-    BL_PROFILE_VAR_STOP(pmain);
+    //BL_PROFILE_VAR_STOP(pmain);
```
[the remark this answers is not in the document]
Comment (ax3l) [reply src/main.cpp]: unrelated, removed.

**Review by cemitch99 — APPROVED** [review]
This looks ok to me.  Q:  Since the naming of elements is optional, why modify every existing example to name every element?

**Comment by ax3l** [comment]
>  Q: Since the naming of elements is optional, why modify every existing example to name every element?

I modified all examples because otherwise the users would not see that naming is possible. There are some tests that I did not modify for advanced usage, and if people want to skip the name, they will see it in the reference guide. But by default, naming things is good practice, so I use it in user-facing examples.

**Pull request merged**